math/src/msm/mod.rs

```diff
@@ -0,0 +1,174 @@
+use crate::cyclic_group::IsCyclicBilinearGroup;
+use crate::field::element::FieldElement;
+use crate::field::traits::IsLinearField;
+
+pub mod naive;
+pub mod pippenger;
+
+pub trait MSM<F, G>
+where
+    G: IsCyclicBilinearGroup,
+    F: IsLinearField,
+{
+    fn msm(&self, ks: &[FieldElement<F>], ps: &[G]) -> G;
+}
```

Comment:
In this new `msm` module we define the `MSM<F,G>` trait, and put all the tests that all implementations should pass.

The tests are the same that were previously used in the previous (now named `naive`) MSM algorithm, plus using `test-case` to target to both implementations (`naive` and `pipppenger`).